### ISSUE TITLE
Access information about object at cursor

### ISSUE THREAD

**Issue by uliska** [wbsoft/frescobaldi#945]
I'm sorry that I have to ask this, but I'm not sure where to start looking.

When hovering over an object or clicking on an object in the music view we have the location of the object in the input flie (and it is displayed in the tooltip).

Where can I start getting more information about that object:

* its grob type
* if it's a note if there are articulations, markup or spanners (or anything else) attached
* if there's a tweak to that object

I don't think I should start parsing the input file myself, this must be provided by `ly.music`, isn't it?

What I want to achieve is a) display more information in a tooltip (starting with an ID (if assigned)) and possibly b) start adding editing functions to the context menu.

Examples for b) could be: 

* add explicit direction to an attached object
* edit the text of a markup
* open a `\shape` editor for a curve

etc.

But initially all I want to do is display the ID in the tooltip

**Comment by uliska** [comment]
Bump: any hints @wbsoft and @PeterBjuhr ?

**Comment by PeterBjuhr** [comment]
Yes, `ly.music` can provide som info. Do you have some example code? It would simplify the discussion, I think...

**Comment by uliska** [comment]
OK, let's take this example: I'm hovering over the slur:

![fresco-point-and-click](https://user-images.githubusercontent.com/1812148/30072015-4369af6e-9269-11e7-8d40-b1e5abb29c1f.png)

The slur is marked by the rectangle, and the tooltip displays file, location, variable name and timing position (in absolute fraction, not as measure). The corresponding input is

![fresco-point-and-click-code](https://user-images.githubusercontent.com/1812148/30072070-792b69da-9269-11e7-8bfe-a76f072fce72.png)

where we can see the slur highlighted. I assume the left one is found as the point-and-click target, and the right parenthesis is highlighted as the matching one (i.e. not directly through point-and-click).

Several things I'd like to retrieve in order to display them in the tooltip:

If there were an override on the slur, like:

```lilypond
  \once \override Slur.id = "one"
  bes8. ^\mf ^\> ( as8 g16 \! ) |
```
I'd like to get the property/value pair.
The same when it is applied as a `\tweak` - although the point-and-click now doesn't point to the slur anymore but to the tweak.

Otherwise, if I'd point at the b flat (we're in a flat, treble clef) I would like to at least get the `\mf`, ideally even the `\>`.

Another thing I'd might want to read is: if I point at the *mf* I'd like to see if it applied using a directional operator.

Some of these would be for informative display in the tooltip, others to start providing editing functionality through the music view (e.g. switching a direction, replacing a dynamic/articulation etc.)

**Comment by PeterBjuhr** [comment]
This the `ly.music` representation of your override example:

```
<Keyword '\\once'>
    <Override '\\override'>
      <PathItem 'Slur'>
      <PathItem '.'>
      <PathItem 'id'>
      <String '"'>
    <Note 'bes'>
      <Duration '8'>
    <Postfix '^'>
      <Dynamic '\\mf'>
    <Postfix '^'>
      <Dynamic '\\>'>
    <Slur '('>
    <Note 'as'>
      <Duration '8'>
    <Note 'g'>
      <Duration '16'>
    <Dynamic '\\!'>
    <Slur ')'>
    <PipeSymbol '|'>
```

**Comment by PeterBjuhr** [comment]
Snippet code to get `ly.music.dump`:
```python
import lydocument
import ly.music

c = lydocument.cursor(cursor)  
lym = ly.music.document(c.document)
                  
text = lym.dump()
```
Just mark a piece of ly source code and apply the snippet.

**Comment by PeterBjuhr** [comment]
I should add that the dump output doesn't cover all data included, so you may find more interesting stuff. I think it could be a good start to take a look into the code, e. g. here is a function for getting the grobname from the override node: https://github.com/wbsoft/python-ly/blob/master/ly/music/items.py#L1158
